src/aws-cpp-sdk-core/source/auth/signer-provider/AWSAuthSignerProviderBase.cpp

```diff
@@ -0,0 +1,10 @@
+/**
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/core/auth/signer-provider/AWSAuthSignerProviderBase.h>
+#include <aws/core/auth/AWSCredentialsProviderChain.h>
+
+std::shared_ptr<AWSCredentialsProvider> AWSAuthSignerProvider::GetCredentialsProvider() const {
+  return MakeShared<DefaultAWSCredentialsProviderChain>("AWSAuthSignerProvider");
```

Comment:
In a perfect world i would want to make the function pure virutal, but for backwards compat we can return the default credential provider.